dEdX/src/dEdxAnalyser.cc

```diff
@@ -571,19 +698,60 @@ void dEdxAnalyser::processEvent( LCEvent * evt ) {
         _MCmomVec.push_back(mcpar0->getMomentum()[1]);
         _MCmomVec.push_back(mcpar0->getMomentum()[2]);
       }
-    }
 
-    // check which hits are inside the TPC
-    TrackerHitVec trkHits = track->getTrackerHits();
-    int nHitsTPC = 0;
-    for(unsigned int iHit=0; iHit<trkHits.size(); iHit++){
-      if(trkHits[iHit]->getType() == 0)
+      // check which hits are inside the TPC
+      TrackerHitVec trkHits = track->getTrackerHits();
+      int nHitsTPC = 0;
+      for(unsigned int iHit=0; iHit<trkHits.size(); iHit++)
       {
-        double x = trkHits[iHit]->getPosition()[0];
-        double y = trkHits[iHit]->getPosition()[1];
-        if(sqrt(x*x+y*y)>=_TPC_inner && sqrt(x*x+y*y)<=_TPC_outer) nHitsTPC++;
+        if(trkHits[iHit]->getType() == 0)
+        {
+          double x = trkHits[iHit]->getPosition()[0];
+          double y = trkHits[iHit]->getPosition()[1];
+          if(sqrt(x*x+y*y)>=_TPC_inner && sqrt(x*x+y*y)<=_TPC_outer) nHitsTPC++;
+        }
       }
-    }
+
+      // if track hits are available check hit spectrum
+      //if(_useHitCol)
+      //{
+        //const TrackerHitVec Hits = track->getTrackerHits();
+        for (unsigned int h=0; h<trkHits.size(); ++h)
```

Comment:
```suggestion
        for (const auto hit : trkHits)
```
And then use `hit` instead of `trkHits[h]`, since you are not using the index `h` anywhere else in this loop it seems.

Reply:
strictly optional.